### PR TITLE
Remove db.connection_string

### PR DESCRIPTION
**Pull request by laurit**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/11069

### CODE REVIEW

File: instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/semconv/db/DbClientCommonAttributesGetter.java [hunk 1]
```diff
@@ -19,6 +19,10 @@ public interface DbClientCommonAttributesGetter<REQUEST> {
   @Nullable
   String getName(REQUEST request);
 
+  // to be removed in the next release, use `server.address` and `server.port` instead
```
Comment (zeitlinger) [inline instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/semconv/db/DbClientCommonAttributesGetter.java:22]: - why in the next release?
- replace "next" with the version number (2.4.0)

**Review by trask — APPROVED** [review]
@breedx-splk do you need more time to adapt to this change, or ok to merge? thanks https://github.com/open-telemetry/semantic-conventions/issues/724#issuecomment-1982010392

**Comment by breedx-splk** [comment]
> @breedx-splk do you need more time to adapt to this change, or ok to merge? thanks [open-telemetry/semantic-conventions#724 (comment)](https://github.com/open-telemetry/semantic-conventions/issues/724#issuecomment-1982010392)

Thanks for checking, appreciate it. Let's proceed.

File: instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/semconv/db/DbClientCommonAttributesGetter.java (hunk 1 above)
Comment (laurit) [inline instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/semconv/db/DbClientCommonAttributesGetter.java:22]:
```suggestion
  // to be removed in 2.4.0, use `server.address` and `server.port` instead
```

**Pull request merged**